tests/test_service.py

```diff
@@ -0,0 +1,368 @@
+        )
+        self.assert_dict_with_keys(actual, ["services"])
+        for service in actual["services"]:
+            self.assertEqual(service["running_port"], 1337)
+            self.assertEqual(service["name"], "telnet")
+            self.assertEqual(service["uuid"], service_uuid_target)
+            self.assertEqual(service["device"], device_uuids[2])
+
+    def test_use_service_not_found(self):
+        with self.assertRaises(ServiceNotFoundException):
+            self.client.ms("service", ["use"], device_uuid=uuid(), service_uuid=uuid())
+
+    def test_use_device_powered_off(self):
+        device_uuid = setup_device(2)[1]
+        service_uuid = create_service(device_uuid)[0]
+        with self.assertRaises(DeviceNotOnlineException):
+            self.client.ms("service", ["use"], device_uuid=device_uuid, service_uuid=service_uuid)
+
+    def test_use_device_not_found(self):
+        device_uuid = uuid()
+        service_uuid = create_service(device_uuid, name="portscan")[0]
+        with self.assertRaises(DeviceNotFoundException):
+            self.client.ms("service", ["use"], device_uuid=device_uuid, service_uuid=service_uuid, target_device=uuid())
+
+    def test_use_permission_denied(self):
+        device_uuid = setup_device(owner=uuid())[0]
+        service_uuid = create_service(device_uuid, owner=uuid(), name="portscan")[0]
+        with self.assertRaises(PermissionDeniedException):
+            self.client.ms("service", ["use"], device_uuid=device_uuid, service_uuid=service_uuid, target_device=uuid())
+
+    def test_use_service_cannot_be_used(self):
+        device_uuid = setup_device()[0]
+        service_uuid = create_service(device_uuid)[0]
+        with self.assertRaises(ServiceCannotBeUsedException):
+            self.client.ms(
+                "service", ["use"], device_uuid=device_uuid, service_uuid=service_uuid,
+            )
+
+    def test_use_invalid_request(self):
+        device_uuid = setup_device()[0]
+        service_uuid = create_service(device_uuid, "portscan")[0]
+        with self.assertRaises(InvalidRequestException):
+            self.client.ms("service", ["use"], device_uuid=device_uuid, service_uuid=service_uuid)
+
+    def test_private_info_successful(self):
+        device_uuid = setup_device()[0]
+        service_uuid = create_service(device_uuid)[0]
+        actual = self.client.ms("service", ["private_info"], device_uuid=device_uuid, service_uuid=service_uuid)
+        self.assert_dict_with_keys(
+            actual, ["running", "owner", "running_port", "name", "uuid", "device", "speed", "part_owner"]
+        )
+        self.assertTrue(actual["running"])
+        self.assertEqual(actual["owner"], super_uuid)
+        self.assertEqual(actual["running_port"], 1337)
+        self.assertEqual(actual["name"], "telnet")
+        self.assertEqual(actual["uuid"], service_uuid)
+        self.assertEqual(actual["device"], device_uuid)
```

Comment:
The first parameter of the `assertEqual` function should be the expected value and the second one the actual value. (applies to other test cases as well)